src/microbe_stage/Compound.cs

```diff
@@ -72,6 +79,14 @@ public void Check(string name)
         TranslationHelper.CopyTranslateTemplatesToTranslateSource(this);
     }
 
+    public void Resolve()
+    {
+        if (!string.IsNullOrEmpty(IconPath))
```

Comment:
I think you should add a check against the Iconpath being null or empty in the Check method, around line 53